src/main/java/seedu/address/model/lesson/Lesson.java

```diff
@@ -33,6 +39,25 @@ public Lesson(LocalTime startTime, LocalTime endTime, Day day, NumberOfOccurrenc
         this.day = day;
         this.numberOfOccurrences = numberOfOccurrences;
         this.venue = venue;
+        this.attendanceRecordList = new AttendanceRecordList(numberOfOccurrences.value);
+    }
+
+    /**
+     * Every field must be present and not null.
+     * Creates a lesson with the specified parameters.
+     */
```

Comment:
Perhaps it is a good idea to keep javadocs consistent?
```suggestion
    /**
     * Every field must be present and not null.
     * Creates a new lesson with the specified parameters.
     */

Reply:
Hmm I wanted to differentiate this constructor from the previous as this constructor is typically not meant to be used to create new lessons, but rather act as a "copy" constructor with modifications. Would it be better to leave it as is, add the word "new", or "Creates a copy of a lesson with the specified parameters."?

Reply:
Would recommend putting new back, since we are technically still generating a new Lesson